### PR TITLE
[Jenkins] Remove low precision OpenCL/GCC tests. Unify stage names. Remove W/A for Vega10 COMGR problem.

### PR DESCRIPTION
**Pull request by atamazov**
### OpenCL backend is not intended to support low precision.

...But takes time on Jenkins. Let's remove some tests from "Fast low precision":
- Half OpenCL/GCC Debug - remove (it takes quite much time to complete)
- Int8 OpenCL/GCC Debug - remove (same reason)
- Half OpenCL/GCC Release - keep
- Int8 OpenCL/GCC Release - keep

### Removed workaround for Vega10 COMGR problem.

The problem was fixed by #529 some time ago.

### Jenkins' stage names unified.

Format: \[_DataType_\] _Backend_[/_Compiler_] _BuildType_ \[_TestSet_\] \[_Target_\]

The only mandatory elements are _Backend_ and _BuildType_; others are optional.

- _DataType_ := `{ Half | BF16 | Int8 | FP32* }`
  - `FP32` is the default and usually not specified.
- _Backend_ := `{ Hip | OpenCL }`
- _Compiler_ := `{ Clang* | hcc |  GCC* }`
  - `Clang` is the default for the `Hip` backend, and implies hip-clang compiler. For the OpenCL backend, "Clang" implies the system x86 compiler.
  - `GCC` is the default for `OpenCL` backend.
  - The default compiler is usually not specified.
- _BuildType_ := `{ Release | Debug [ BuildTypeModifier ] }`
  - _BuildTypeModifier_ := `{ COMGR | Embedded | Static | Package | Npormal-Find | Fast-Find | Tensile | Tensile-Latest | Package | ... }`
- _TestSet_ := `{ All | Subset | Smoke* }`
  - `All` corresponds to `cmake -DMIOPEN_TEST_ALL=On`.
  - `Subset` corresponds to _Target_- or _BuildTypeModifier_-specific subsetting of the `All` testset, e.g. `-DMIOPEN_TEST_GFX908=On` or `-DMIOPEN_TEST_MIOTENSILE=On`.
  - `Smoke` (`-DMIOPEN_TEST_ALL=Off`) is the default and usually not specified.
- _Target_ := `{ gfx908 | Vega20 | Vega10 | Fiji | Vega* } `
  - `Vega` (gfx906 or gfx900) is the default and usually not specified.

### CODE REVIEW

File: Jenkinsfile
Comment (ce1adon) [inline Jenkinsfile:317]: so the comgr test is no longer restricted to vega20?

**Review by ce1adon — DISMISSED** [review]
LGTM

File: Jenkinsfile
Comment (atamazov) [reply Jenkinsfile:317]: Yes. That was a bug that prevented COMgr to work with Vega10 correctly. Fixed by #529

**Review by DrizztDoUrden — DISMISSED** [review]

**Comment by atamazov** [comment]
This is preview of how "unified" names of test steps will look like after this change:
image

**Comment by atamazov** [comment]
@DrizztDoUrden Please re-approve (after merge).

@daniellowell All CI tests passed, this can be merged now.

**Review by DrizztDoUrden — APPROVED** [review]

**Pull request merged**
